### ISSUE TITLE
Probable issue/bug: Conflict with FPI + Tracking Protection

### ISSUE THREAD

**Issue by EstherMoellman** [erosman/support#153]
Hi @erosman ! Hope everything is fine with you.

I believe I found a kind of issue at FireMonkey. Of course, I apologize in advance if I did a mistake. I tested a lot before opening this issue, but even if the mistake is on my side, I think is worth sharing with you.

Basically the issue happens when "privacy.firstparty.isolate" is TRUE, about:preferences => privacy & security => cookies (block all cookies)... if a website is white-listed (allowing cookies) then FireMonkey seems to create a kind of conflict, where the white-listed webpage is not recognized, cookies seem to be blocked (despite Nightly shows the webpage as white-listed), and only by blacklisting and white-listing the webpage again cookies will be recognized again (but only for one time, it will be necessary to redo the procedure every-time a white-listed webpage is loaded in new tab).

Sadly, this issue is not easy to reproduce, is not happening all the time. That's why took me lot of tests, because it's a kind of random issue. But basically, these are the steps to reproduce:

1) Do a fresh Nightly' install + restart twice.

2) At about:config => privacy.firstparty.isolate: TRUE => Restart.

3) At about:preferences => Privacy and Security:
Block all cookies
Set Nightly to delete everything at close (VERY IMPORTANT: Every-time Nightly' launches, must to be totally cache clean, cookies deleted etc. Otherwise, the issue won't appear because will use stored stuff).

4) Restart Nightly.

5) Load the following webpages, and white-list them (At MenuBar => mouse-click shield icon (Enhanced Tracking Protection on the left of UrlBar) => A popup will open => Turn off the main button => Confirm by seeing broken shield icon at MenuBar saying the webpage has Enhanced Tracking Protection OFF). If all cookies are blocked, these webpages are not going to load. They only will load if are white-listed at Enhanced Tracking Protection:
https://imgur.com/
https://userstyles.org/
https://outlook.live.com/owa/ (here you also will need to white-list https://login.live.com/cookiesDisabled.srf? and in order to test this page, you must click "Sign in").
https://www.imdb.com/title/tt8946378/ (the test here will be by mouse-clicking the video. If Enhanced Tracking Protection is ON, the video will be blocked - because needs cookies. White-list the webpage, and the video will work).

Why 4 webpages? Because as I explained you above, it is not easy to reproduce the issue.
And these 4 pages only work when Enhanced Tracking Protection is OFF.
AGAIN, VERY IMPORTANT: imgur.com and usertyles.org are easy to test. But in order to test:
- https://outlook.live.com/owa/ (you need to click "Sign in")
- https://www.imdb.com/title/tt8946378/ (you need to click the video)

6) Restart => Check everything is working fine, webpages are white-listed and loading without problems.

7) Do a Firemonkey clean install => Restart.

8) Create a new CSS inside Firemonkey. For example:

```
/*
==UserCSS==
@name         12bytes
@match         *://*.12bytes.org/*
@run-at         document-start
==/UserCSS==
*/

.post-thumbnail { background: transparent !important }
```

PS: You don't need a specific CSS/JS' scripts in order to see the issue. You can use any CSS/JS' script with the 4 webpages I pointed you. There is no relation between the CSS/JS' script and the 4 webpages. Any CSS/JS' script will trigger the issue I reported you, if you use the 4 webpages I mentioned.

9) Restart and ensure Nightly will clean cache, cookies and everything.

10) Test one by one each webpage at point "5)" above. It's important not to use pinned tabs, nor saved session etc. Test each webpage from the scratch, in a new tab, no cache, no cookies etc. You may see the following possibilities:
- One of the webpages is not loaded. Even if white-listed (and icon at NavBar confirms this), cookies are blocked. If you blacklist and white-list again, the webpage may work, but you will need to do that every-time you load the webpage with cache/cookies cleaned.
- Another possibility is that you will see more than one webpage not working, same reason, same process, same previous explanation.
- But if all the webpages are loading, then in order to reproduce the issue, please create another CSS inside Firemonkey. Generally with just one CSS' script, the issue already appears at least for one webpage (mentioned above). But sometimes all the 4 tested webpages are loading, and in this case the only way to reproduce the issue, is to start adding CSS and JS' scripts. At some point, with one CSS' script or more CSS/JS' scripts, the issue will appear, and webpages are not going to load.

Again, it seems a kind of conflict, where for an unknown reason to me, when Firemonkey has at least one CSS or JS' script, somehow is interfering with Enhanced Tracking Protection + FPI, Nightly can't identify that the webpage is white-listed (even if the icon shows the webpage as white-listed), cookies appear to be blocked, and the webpage doesn't load.
As I mentioned, in this case, if the webpage is blacklisted and white-listed again, it'll be work. Sadly, the issue will appear immediately the next time the webpage is loaded in a new tab (no cache, no cookies).

If you keep Firemonkey enabled, but without CSS/JS' scripts... the issue is not appearing at all.
Also, if you keep Firemonkey enabled and with hundreds of CSS/JS' scripts, but at about:preferences you change to "block all third-party cookies"... the issue is not appearing at all. The issue appears only when "block all cookies" is set.
Also, disabling  privacy.firstparty.isolate: FALSE... the issue is not appearing at all.

Thank you in advance, have a nice weekend, and big hug! : )

**Comment by erosman** [comment]
> I believe I found a kind of issue at FireMonkey. Of course, I apologize in advance if I did a mistake. 

No problem :)

FireMonkey does not have any dependencies on the cookies enabled or not. 

However, nowadays, some sites insist on cookies and server a different version when cookies are blocked. That has nothing to do with FireMonkey.

I tested it and it is working.
I also tested it with all cookies blocked and it was still working.

Please note:
Firefox caches the injected JS/CSS. If you make changes, and you have 2 tabs open for `12bytes.org `and refresh one, it will reload the same old JS/CSS from FireMonkey. Close both tabs and open a new tab to test.

**Comment by EstherMoellman** [comment]
@erosman , thank for your answer, as usual promptly and helpful reply.

First thinks first: If you exactly followed my steps, and you can't reproduce my issue, and everything is working fine to you... well, you should close this issue.

Sadly, from my side, the issue is still there.
Even if I have a single tab for each of the 4 webpages I pointed you, following exactly the steps I described in my previous message, I can reproduce the issue in my computer and different computers.

My tests were done with Nightly' clean fresh install and Firemonkey clean fresh install. And the issue appears after following exactly the steps I described. So, if I did a mistake, is very hard to me to identify where am I mistaking.

Thank you anyway.

PS: I did same tests with Stylus and other similar add-ons, I don't have the issue. It only appears to me when I use Firemonkey, at least with one or more CSS/JS' scripts.

**Comment by EstherMoellman** [comment]
PS: By the way, imgur.com and usertyles.org are easy to test. But just remembering that in order to test:

- https://outlook.live.com/owa/ (you need to click "Sign in")

- https://www.imdb.com/title/tt8946378/ (you need to click the video)

**Comment by erosman** [comment]
The sample CSS you posted was for 12bytes and that was the only one I tested.
I tested with latest Nightly.

**Comment by EstherMoellman** [comment]
... sorry if I wasn't clear!

You don't need specific CSS/JS' scripts in order to see the issue. You can use any CSS/JS' script with the 4 webpages I pointed you.
At point "8)" in my previous message, I posted a CSS' script for 12bytes just as an example, but you can use any CSS/JS' script you want (with the 4 webpages I tested).
There is no relation between the CSS/JS' script and the 4 webpages.
Any CSS/JS' script will trigger the issue I reported you, if you use the 4 webpages I mentioned.

Just remember my last post: imgur.com and userstyles.org are easy to test. But in order to test:

- https://outlook.live.com/owa/ (you need to click "Sign in")

- https://www.imdb.com/title/tt8946378/ (you need to click the video)

The only about:config pref you need to change is privacy.firstparty.isolate: TRUE. That's all.
There is no way to use First Party Isolation through "about:pages", only through about:config. This pref is totally safe, and without problems I've been using it for a long time with FireMonkey.

**Comment by erosman** [comment]
Test CSS

```CSS
/*
==UserCSS==
@name           12bytes
@match          *://12bytes.org/*
@match          *://imgur.com/
@match          *://outlook.live.com/owa/*
@match          *://userstyles.org/
@match          *://www.imdb.com/title/tt8946378/*
@run-at         document-start
==/UserCSS==
*/


body { border: 2px solid red; }
```

Tested with
`privacy.firstparty.isolate`: TRUE
about:preferences -> Privacy and Security -> Block all cookies

- Tested `https://12bytes.org/`  CSS injected fine
- `https://imgur.com/` CSS injected fine but page does not load
- `https://userstyles.org/` CSS injected fine but page does not load
- `https://outlook.live.com/owa/` CSS injected fine
- 'https://www.imdb.com/title/tt8946378/'  CSS injected fine

> https://www.imdb.com/title/tt8946378/ (the test here will be by mouse-clicking the video. If Enhanced Tracking Protection is ON, the video will be blocked - because needs cookies. White-list the webpage, and the video will work).

The issue listed related to Firefox. FireMonkey is injecting the CSS.

**Comment by EstherMoellman** [comment]
Thank you again @erosman .

Well, I'm happy you reproduced the issue.
Now, some important comments... as I wrote in my first comment: If you disable FireMonkey... the issue is gone. Even if you enable FireMonkey, but without CSS/JS' scripts... the issue also is gone. In brief: The issue only appears when I use FireMonkey not empty, with one or more CSS/JS' scripts.

No, I'm not saying this necessarily is a FireMonkey' bug.
Yeah, it could be a Nightly' bug.
But, yes, I'm saying that this is only happening with FireMonkey, and strangely, it happens only when FireMonkey has one or more CSS/JS' scripts (even when the scripts are not related to the webpages).

And yes, First Party Isolation may break webpages. But this happened two years ago. Nowadays is very rarely FPI to break webpages. And the 4 webpages I tested, are not breaking with FPI. Just disable FireMonkey, and you'll see the 4 webpages work perfectly. The issue appear only withe FireMonkey.

This issue may be a Nightly' bug, but is 100% related only to FireMonkey.
Now you reproduced it, I'm totally sure, this is a bug. but it has relation with Nightly/Firemonkey.
Do you want me to open a Nightly' bug? Or before that you want to investigate more?

**Comment by erosman** [comment]
> Well, I'm happy you reproduced the issue.

Actually that is not the case.. FireMonkey is injecting the CSS into all the above mentioned page. 

Let me repeat:

- https://12bytes.org/ CSS injected fine
- https://imgur.com/ CSS injected fine, page does not load with FireMonkey enabled or disabled
-  https://userstyles.org/ CSS injected fine but page does not load with FireMonkey enabled or disabled
- https://outlook.live.com/owa/ CSS injected fine
- https://www.imdb.com/title/tt8946378/ CSS injected fine

As you can see, FireMonkey has no bearing on page loading or not. Those pages do not load even if FireMonkey has been disabled.

AFA FireMonkey injecting CSS, it is doing it in all those pages. In other words, the issue you are experiencing does not relate to FireMonkey (unless you are injecting some other scripts that breaks the page).

It is your setting that is breaking the page.

**Comment by EstherMoellman** [comment]
It's not possible to be just a matter of "my settings", because if I disable FireMonkey, everything works perfects.
So, if you're having results "X", and I'm having results "Y", this is because you and I, we're testing stuff in different ways.
I'm almost sure that if you reproduce exactly my steps, you'll have same results. But my steps should be reproduced exactly (otherwise you'll have different results). I know my first post was long and boring. But I'm convinced that your test was not exactly my test. It took me a week to discover details about how to reproduce the issue. It has details, and steps must strictly be followed if you want to see the issue.

Again, at my tests if I disable FireMonkey, everything works fine.
If I do not disable FireMonkey and leave it empty without scripts, also everything works fine.
Also, If I use "my settings", but without FireMonkey, everything works fine (see point "5)" above at my first post. The 4 tested webpages must be whitelisted).

So, it's not just "my settings".
It's something between Nightly' settings and FireMonkey. Something there in the middle is conflicting. I don't know what or why, but something is conflicting.
The issue only appear when I insert one or more scripts at FireMonkey.
I can send you a video if you want. I tested in different computers, and almost took me a week of tests.

But @erosman if you still believe this has nothing to do with FireMonkey, and this is a Nightly' bug, I'm not going to insist you. Please, in this case feel free to close the issue.

**Comment by erosman** [comment]
> The issue only appear when I insert one or more scripts at FireMonkey.

Could it be the script themselves that cause issues?

In any case, I cant reproduce on 75.0a1 (2020-02-28) (64-bit) Windows so I cant investigate it.

Which OS are you on?

There are peculiarities with some OS like I heard issues with Safari (e.g.: `https://github.com/Tampermonkey/tampermonkey/issues/876`).

**Comment by EstherMoellman** [comment]
> 
> 
> Could it be the script themselves that cause issues?
> 

As I explained in detail at my first post, it doesn't matter which script is used.
FireMonkey disabled => Everything works perfect.
FireMonkey enabled but empty without scripts => Everything works perfect.
FireMonkey with one or more scripts, any scripts => At least one of the 4 webpages won't work, generally 2 of them won't work.
You can use any script, even if it is not related to the 4 webpages. The issue appears at the moment one script is inserted at FireMonkey.

> 
> Which OS are you on?
> 

Windows 10 (1909) fully updated.
Latest Nightly, clean install.
But I tested in several different devices, and the issue always appear (if steps described in my first post are followed).

**Comment by erosman** [comment]
You can ask people at Bugzilla. They know more about the effect of changes made to the `about:config`

**Comment by EstherMoellman** [comment]
My friend, our conversation is starting to be repetitive. Perhaps is better to you to close this issue.

I do appreciate all your intention for helping me.
But with all due respect, again, this issue is not related just to Nightly' settings, about:config etc.
I understood in the past you have a discussion with another user, using wrong FF' preferences. But definitely this is not my case. With my FF' settings the 4 webpages work perfectly.

3 things together are triggering the issue I posted here:
1) First Party Isolation
2) Block all cookies
3) FireMonkey

If I disable FireMonkey, everything works fine.

So, again, the problem is not First Party Isolation, nor Block all cookies, nor about:config, nor "my settings" etc.
By enabling FireMonkey, that the issue appears.
If I change to Stylus or other add-on, everything works perfect.

But again @erosman, really I thank you for your time and help.

Have a nice Sunday and big hug! : )

**Comment by erosman** [comment]
Can you test with only this CSS and NOTHING else (no other scripts/CSS)?

```CSS
/*
==UserCSS==
@name           12bytes
@match          *://12bytes.org/*
@match          *://imgur.com/*
@match          *://outlook.live.com/owa/*
@match          *://userstyles.org/*
@match          *://www.imdb.com/title/tt8946378/*
@run-at         document-start
==/UserCSS==
*/


body { border: 2px solid red; }
```

**Comment by EstherMoellman** [comment]
Good morning @erosman ,

I tested your CSS (nothing else, no other scripts), and the result was exactly the same as mine yesterday, with your CSS (nothing else) two webpages are not fully working:

- outlook.live.com when "Sign in" is clicked, cookies are blocked (despite webpage is whitelisted at Enhanced Tracking Protection, and shield icon is broken confirming white-list), and the login page is not loaded.

- same goes to www.imdb.com/title/tt8946378 when the video is clicked, cookies are blocked (despite white-listed at ETP), and video doesn't load.

It is exactly what I explained yesterday.

And if I disable your CSS' script, but with FireMonkey enabled, everything works perfect.
FireMonkey enabled, but without scripts => Everything works perfect.
And of course, FireMonkey disabled => Everything works perfect.

When I enable FireMonkey, and I enable your CSS' script (or any CSS/JS' script, one or more scripts, even scripts not related to the 4 tested webpages)... then the issue appears, and webpages are not fully working (cookies blocked when they suppose to be allowed, because they are white-listed at ETP).

By enabling any script at FireMonkey triggers a kind of conflict with Nightly, where ETP can't work properly, permissions are not properly recognized, and everything remains blocked (not just cookies).

From my ignorance, it is like when a CSS/JS' script is enabled at FireMonkey, if FirstPartyIsolation is enabled and Block all cookies is enabled at FF, then webpages are not recognized as white-listed at Enhanced Tracking Protection. Cookies are supposed to be allowed for the white-listed webpages, the broken shield icon is confirming that webpages are whitelisted, but at the precise moment I enabled one or more CSS/JS' scripts at FireMonkey, any script, even not related to specific webpage... then the issue appears.
From my ignorance, my wild guess is that there is conflict between the API and the permissions.sqlite file. I believe this is a Nightly' bug, but triggered by FireMonkey.
Stylus and similar add-ons are working well, because they don't use FireMonkey' API.
This is my ignorant opinion.

Until 15 days ago, everything was working perfect between Nightly, FirstPartyIsolation, Block all cookies and FireMonkey. I've being using this combination along months, without problems. This conflict started days ago, and I'm sure that it was due to a Nightly' update. I'm also sure that this problem is not exclusively triggered by FireMonkey, with time we are going to see other Nightly' users complaining about the same issue, but with other add-ons using the same or similar API.

**Comment by erosman** [comment]
I re-tested with `https://imgur.com/`

`privacy.firstparty.isolate: true`
about:preferences => Privacy and Security -> Block all cookies
The above mentioned script
Deleted cache & cookies 
Restarted
tested with **normal tab**, **container tab**, & **Private window**

Page does not fully load but CSS is injected without problem

Next ...
Enhanced Tracking Protection is OFF for this site
Page does loads and CSS is injected without problem


What I get is that:
FireMonkey works (injects CSS) and has no effect on the out come of page loading or not

When page doesn't load, the base HTML/Javascript is there but due to isolation, content can not be retrieved form 3rd party sources.

**Without white-listing:**
The following are the errors and the consequence of isolation & blocking cookies

```
Request to access cookie or storage on “<URL>” was blocked because we are blocking all storage access requests. 3
Loading failed for the <script> with source “https://www.googletagmanager.com/gtag/js?id=UA-6671908-15”. imgur.com:1:1
Loading failed for the <script> with source “https://www.googletagmanager.com/gtm.js?id=GTM-M6N38SF”. imgur.com:1:1
SecurityError: The operation is insecure. main.536443864319279e62f3.js:79
```

None of the errors relate to FireMonkey and its operation. In fact the userScript API injects into a sandbox and doesn't affect the page loading at all.

There are still errors with `imgur.com` even after white-listing:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://api.amplitude.com/. (Reason: CORS request did not succeed).
Loading failed for the <script> with source “https://ced.sascdn.com/tag/1357/smart.js”. imgur.com:1:1
Loading failed for the <script> with source “https://ced-ns.sascdn.com/diff/templates/ts/dist/viewability/sas-viewability-1.1.js”. imgur.com:1:1
```

To clarify, this it the content of `https://imgur.com/`
It is exactly the same with cookies or without cookies, with isolation or without, with FireMoneky or without.

The rest of the content is retired with JavaScript from remote sources. If cookies are blocked and `firstparty.isolate`, it prevents the browser to get that content and display it.

The above mentioned operation is handle by the browser and FireMonkey has not part in it.

```HTML
<!doctype html> <html lang=en> <head> <meta charset=utf-8> <meta name=viewport content="width=device-width,initial-scale=1"> <title>Imgur: The magic of the Internet</title> <meta name=keywords content="funny, image, gif, gifs, memes, jokes, image upload, upload image, lol, humor, vote, comment, share, imgur, imgur.com, wallpaper"/> <meta name=description content="Discover the magic of the internet at Imgur, a community powered entertainment destination. Lift your spirits with funny jokes, trending memes, entertaining gifs, inspiring stories, viral videos, and so much more."/> <meta name=copyright content="Copyright 2019 Imgur, Inc."/> <meta name=monetization content=$ilp.gatehub.net/472322412> <link rel=icon type=image/png href=https://s.imgur.com/images/favicon-32x32.png sizes=32x32> <link rel=icon type=image/png href=https://s.imgur.com/images/favicon-96x96.png sizes=96x96> <link rel=icon type=image/png href=https://s.imgur.com/images/favicon-16x16.png sizes=16x16> <link rel=apple-touch-icon-precomposed href=https://s.imgur.com/images/favicon-152.png> <meta name=msapplication-TileColor content=#2cd63c> <meta name=msapplication-TileImage content=https://s.imgur.com/images/favicon-144.png> <meta property=og:url content=https://imgur.com/ data-react-helmet=true /> <link rel=alternate media="only screen and (max-width: 640px)" href=https://m.imgur.com/ > <meta name=p:domain_verify content=834554521765408b9effdc758b69c5ee /> <meta property=og:site_name content=Imgur /> <meta property=fb:admins content=12331492 /> <meta property=fb:admins content=12301369 /> <meta property=fb:app_id content=127621437303857 /> <meta property=al:android:url content="imgur://imgur.com/?from=fbreferral"/> <meta property=al:android:app_name content=Imgur /> <meta property=al:android:package content=com.imgur.mobile /> <meta property=al:ios:url content="imgur://imgur.com/?from=fbreferral"/> <meta property=al:ios:app_store_id content=639881495 /> <meta property=al:ios:app_name content=Imgur /> <meta property=al:web:url content=https://imgur.com/ /> <meta name=twitter:site content=@imgur /> <meta name=twitter:domain content=imgur.com /> <meta name=twitter:app:id:googleplay content=com.imgur.mobile /> <meta name=twitter:card content=summary_large_image /> <meta property=og:type content=website /> <meta property=og:title content=Imgur /> <meta name=twitter:title content=Imgur /> <meta property=og:description content="Imgur: The magic of the Internet"/> <meta name=twitter:description content="Imgur: The magic of the Internet"/> <meta property=og:image content=https://s.imgur.com/images/logo-1200-630.jpg?2 data-react-helmet=true /> <meta property=og:image:height content=630 data-react-helmet=true /> <meta property=og:image:width content=1200 data-react-helmet=true /> <meta name=twitter:image:src content=https://s.imgur.com/images/logo-1200-630.jpg?2 data-react-helmet=true /> <meta name=twitter:image:height content=630 data-react-helmet=true /> <meta name=twitter:image:width content=1200 data-react-helmet=true /> <script>dataLayer=[];var pbjs=pbjs||{};pbjs.que=pbjs.que||[]</script> <script>!function(e,t,a,n,g){e[n]=e[n]||[],e[n].push({"gtm.start":(new Date).getTime(),event:"gtm.js"});var m=t.getElementsByTagName(a)[0],r=t.createElement(a);r.async=!0,r.src="//www.googletagmanager.com/gtm.js?id=GTM-M6N38SF",m.parentNode.insertBefore(r,m)}(window,document,"script","dataLayer")</script> <link href="https://s.imgur.com/desktop-assets/css/styles.c4bb87a94f3ff0fce97a.css" rel="stylesheet"></head> <body> <noscript><iframe src="https://www.googletagmanager.com/ns.html?id=GTM-M6N38SF" height=0 width=0 style=display:none;visibility:hidden></iframe></noscript> <noscript>If you're seeing this message, that means <strong>JavaScript has been disabled on your browser</strong>, please <strong>enable JS</strong> to make Imgur work. </noscript> <div id=root></div> <script async src="https://www.googletagmanager.com/gtag/js?id=UA-6671908-15"></script> <script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","UA-6671908-15",{send_page_view:!1})</script> <script type="text/javascript" src="https://s.imgur.com/desktop-assets/js/main.536443864319279e62f3.js"></script></body> </html> 
```


I have no idea why the result you get with FireMonkey and without is not the same as mine. FireMonkey, or other userScript mangers, play no part in browser scope operations of getting remote content in this case.

**Comment by EstherMoellman** [comment]
@erosman , I believe you are having different test results, because your tests are not exactly my tests (as I detailed in my first post).

Again, it took me a week to find a way to reproduce the issue. So, I'm not surprised you're having different results.
This issue is related to 2 sources (Nightly and FireMonkey), and at the same time also is related to e variables (FirstPartyIsolation, Block all cookies and CSS/JS' scripts at FireMonkey). So, you have more than 30 scenarios, and the issue appears only at 1 scenario (detailed in my first post).
The slightest difference between your test and my test, will give different results.
And different test results, give different conclusions.
That's the reason you're concluding this is a Nightly' issue, or "my settings" issue, or "about:config" issue, or "FPI" issue etc. Again, this your conclusion is due to your tests.

Therefore, the only way to change your conclusions, is by changing your tests.
Only if you reproduce, step by step, the test I detailed at my first post, that we're going to be able to compare results.
Otherwise, we're going to lose time.

For example, at my first post I explained that every test should be done with the 4 webpages.
Also, I explained what to do in each case, to reproduce the issue.
Yeah, my first post in too long, boring, wrote in bad English etc. I understand. But if you don't follow my steps... you're going to have different tests => different conclusions.
It's not going to work the way you're testing.
You're trying to do what I did, but you're trying to do that in your way, your steps etc. I totally understand you, you're trying to quick thinks, you're trying to discover quickly what is going on etc. It's normal. But sadly, you're ending doing your own tests, on your own steps.
And this is giving you different results => different conclusions.

We're losing time.

I can't change your conclusions.
I can't change your tests.
I only can invite you to reproduce exactly the steps I detailed at my first post. If you don't want to that, then please close the issue... it's fine, I totally understand you.

From my side, I only can confirm you that:
- The steps I described at my first post, can reproduce the issue in different devices and different users
- If FireMonkey is disabled, the 4 webpages work perfectly (with FirstPartyIsolation, Block all cookies, Enhanced Tracking Protection white-listed).
- If FireMonkey is enabled, but empty (no scripts), everything also works perfect.

The last think I can offer you is a video.
In a video I can show you everything I already reported to you.
But again, in my opinion we're losing time.
Our problem (yours and mine) is the fact you and I are having different tests. As long as you and I are not having same tests, we always are going to have different results => different conclusions.
So perhaps is better to close this issue.

It's important to stress that regardless our disagreement, I'm extremely grateful to you for all the time and attention to invested in this issue.

**Comment by erosman** [comment]
> can reproduce the issue in different devices and different users

Is anyone else experiencing this issue?

**Comment by EstherMoellman** [comment]
> 
> Is anyone else experiencing this issue?

Yeap!

If you remember, I'm part of a kind of small club, around 150 users from 8 different countries, having as hobby the Nightly' customization (CSS, JS, user.js, preferences etc), and also exchanging add-on experiences between us.

Most of these "club members" are "average users", they don't have technical background, no know-how about tech or IT' stuff. They just enjoy Nightly' customization.
But few of these "club member", around 10 users, they're good enough for helping me to test stuff. And they helped me to test this issue I'm reporting you.

However, as I explained at my first post, this is a tricky issue, is not easy to reproduce, it demands lots of steps and details. So from 10 users, 7 were not capable to reproduce the issue. But 3 of these users are perfectly reproducing the issue.

Every-time I try to understand why the 7 users had different results, I found the same I found in your case: They're doing their own tests. It's normal and I understand. They're intuitively trying to understand the issue under their logic and knowledge. So they jump my test steps. Sadly, they don't understand that by changing tests, they have different results, which drove to different conclusions.

And I can't force people to follow my test steps.
It's not nice from mu side.
I don't like to force.

Yesterday I opened a bug at bugzilla.
Based on my experience, I kind of know what it's going to be the answer: They're going to blame FireMonkey, or they are going to say that can't reproduce the issue. I saw this many times in the past.

Probably future Nightly' updates will correct this bug.
For many months my Nightly + FPI + Blocked all cookies + FireMonkey worked like a charm. So, I believe in the future new Nightly' updates will solve this issue.

**Comment by erosman** [comment]
Let's see if they have any insight in the issue. At least they know what has changed recently.

**Comment by EstherMoellman** [comment]
I did a "detective job" by testing one by one each Nightly' version from the latest to the oldest, and I found that Nightly 02/24/2020 (https://download-origin.cdn.mozilla.net/pub/firefox/nightly/2020/02/2020-02-24-09-47-51-mozilla-central/firefox-75.0a1.en-US.win64.installer.exe) has not the bug I reported.

I tested Nightly 02/25/2020 and it has the bug I reported.
So, it seems the bug appeared at Nightly between day 02/24/2020 and day 02/25/2020.

I already added this info at bugzilla.

**Comment by EstherMoellman** [comment]
@erosman , in case you're interested, I did a video:

https://youtu.be/thhUQ2tjVTo

**Comment by erosman** [comment]
Thank you. I will check it out.

**Comment by EstherMoellman** [comment]
@erosman , if you want to follow, here is the reported bug at bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1619091

They already reproduced the issue, and now the bug report is confirmed.

**Comment by erosman** [comment]
Let's hope they can find the cause and if there is something I can do to fix it, I will do.

**Comment by erosman** [comment]
**Off-topic**: I wonder why the pop up in your video look different I mean the new script/css buttons :thinking: 

![image](https://user-images.githubusercontent.com/6458876/76339796-d2c4e600-630f-11ea-9b5a-2da84d9e708f.png)


It should look like this:

![image](https://user-images.githubusercontent.com/6458876/76339975-128bcd80-6310-11ea-918e-e7a9bc5158cb.png)

**Comment by EstherMoellman** [comment]
upsss, I never paid attention to this detail.
I have no idea why is different. Perhaps due to FireMonkey' icon position? Perhaps Nightly has a kind of "auto-adjustment", enlarging or shortening popups according icons position or other elements? Or perhaps may be the screen size/definition? Or GPU' rendering?
All I can say is that I did my video with a Nightly' and a FireMonkey both clean install.

**Comment by erosman** [comment]
It could be some changes to the `userContent.css` :thinking: 

BTW... I have some news for you. Next v1.31 will have Dark theme for the code area :smile:

**Comment by EstherMoellman** [comment]
... yeah. FM' Dark Theme is good news for me... thanks @erosman !

But with regards to the video, as I mentioned, there I used a totally Nightly + FireMonkey' clean install... so no userContent.css at all... nothing, everything untouched. My ignorant guess again, is that the difference might be related to screen resolution, GPU & Rendering or some other hardware factor. Anyway, This difference may be mysterious, but is not an issue at all : )

**Comment by erosman** [comment]
v1.31 uploaded

You can check to see if it made any difference as the code has changed in some sections.
You can also check the Dark theme and let me know if there are any feedback (I dont use Dark themes).

**Comment by EstherMoellman** [comment]
The good news is that the new FM' Dark Theme rocks.
I don't use your colors for my Dark Themes, but don't worry, you never will please "color personal tastes", everyone has its preferred colors.
The important thing here is that your FM' Dark Theme covered all the "whites", and you did that in a harmonious aesthetic way. Also, buttons and elements are perfectly visible. So, mission accomplishment! Good job! Congrats.

The bad news is that the bug... is still there : (

**Comment by erosman** [comment]
> The bad news is that the bug... is still there : (

:man_facepalming:

Let's hope Bugzilla comes up with some answers.

**Comment by EstherMoellman** [comment]
Hi @erosman , off-topic and with regards to Dark Theme: As you know, at the pop up both scripts (CSS/JS) have on the left-side a "check" symbol (✓). But this symbol appears with the same color if scripts are active or inactive. A script if disabled is grayed (which is perfect). But the "✓" doesn't change color (which may be confusing). Suggestion 01: In the name of isonomy, keep everything with same colors: Orange color for active scripts and active symbol "✓", and gray for disabled script and disabled "✓" symbol. Suggestion 02: Eliminate "✓" symbol, is redundant, the orange script color already informs everything is enabled, and grey color informs disabled. No need for "✓" symbol. And the less, the better! : )

**Comment by erosman** [comment]
I forgot to change the disabled state  :stuck_out_tongue: 

Thank you for letting me know. 
It will be in the next update.

**Comment by erosman** [comment]
I saw Luca getting involved in the bug report.

FYI ... He is the person who created the userScripts API (although in this case FireMonkey is using contentScripts API for injecting css).

**Comment by EstherMoellman** [comment]
... yeah, I saw too... from my ignorance, it seems that there the "key guy" is "Ehsan Akhgari", my guess is that on past 02/24/2020, Ehsan changed something by trying to fix a bug, and unintentionally Ehsan' fix broke something, creating the bug I discovered. If my guess is correct, Ehsan' collaboration may solve the bug very quickly. But Ehsan was called twice, and he is not answering/appearing.

Meanwhile my workaround has been to allow cookies, but to mitigate tracking I activated Fission (in order to isolate cross-site iframes). The problem is that Fission also has bugs.
